pax-web-jetty/src/main/java/org/ops4j/pax/web/service/jetty/internal/JettyServerWrapper.java

```diff
@@ -244,28 +232,18 @@ void removeContext(final HttpContext httpContext) {
 			if (context == null) {
 				return;
 			}
-			int nref = context.decrementRefCount();
-			if (nref <= 0) {
-				try {
-					readLock.unlock();
-					writeLock.lock();
-					LOG.debug(
-							"Removing ServletContextHandler for HTTP context [{}].",
-							httpContext);
-					context = contexts.remove(httpContext);
-				} finally {
-					readLock.lock();
-					writeLock.unlock();
-				}
-			} else {
-				LOG.debug(
-						"ServletContextHandler for HTTP context [{}] referenced [{}] times.",
-						httpContext, nref);
-				return;
-			}
-		} finally {
-			readLock.unlock();
-		}
+                } finally {
+                        readLock.unlock();
+                }
+			
+                try {
+                        writeLock.lock();
+                        LOG.debug("Removing ServletContextHandler for HTTP context [{}].", httpContext);
+                        context = contexts.remove(httpContext);
+                } finally {
+                        writeLock.unlock();
+                }
+                
 		// Destroy the context outside of the locking region
```

Comment:
Need to investigate why and when we introduced this. It's been a multithreading thing, afaik it's been related to race conditions while un-/deploying applications a couple times. There is a test for that race condition.